app/pages/challenges/components/challengeForm.tsx

```diff
@@ -24,9 +34,86 @@ export const ChallengeForm: React.FC<ChallengeFormProps> = ({ userId }) => {
     goal: "",
     unit: "",
   });
-  const [friendIds, setFriendIds] = useState<string>("");
-  const [invitedFriends, setInvitedFriends] = useState<string[]>([]);
+  const [invitedFriends, setInvitedFriends] = useState<FriendOption[]>([]);
   const [errors, setErrors] = useState<Record<string, string>>({});
+  const [isSearchOpen, setIsSearchOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const searchRef = useRef<HTMLDivElement>(null);
+  const [debouncedSearchTerm, setDebouncedSearchTerm] = useState("");
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDebouncedSearchTerm(searchTerm);
+    }, 300);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [searchTerm]);
+
+  const friendIdsQuery = useQuery({
+    queryKey: ["friendIds", userId],
+    queryFn: async () => {
+      const response = await apis().friends.getFriends(userId);
+      if (response.status !== 200) {
+        throw new Error(`Error fetching friends: ${response.statusText}`);
+      }
+      return response.data as number[];
+    },
+    enabled: !!userId,
+  });
+
+  const friendProfilesQuery = useQuery({
+    queryKey: ["friendProfiles", friendIdsQuery.data],
+    queryFn: async () => {
+      if (!friendIdsQuery.data) return [];
+
+      const friendProfiles = await Promise.all(
+        friendIdsQuery.data.map(async (id) => {
+          try {
+            const profileResponse = await apis().user.getUserProfile(id);
+            return {
+              id: profileResponse.data.id,
+              name: profileResponse.data.name,
+              imageUrl: profileResponse.data.imageUrl,
+            };
+          } catch (error) {
+            console.error(`Failed to fetch profile for user ${id}:`, error);
+            return null;
+          }
+        }),
+      );
+
```

Comment:
extract those hooks into seperate hook file . why ? because they could be reused somewhere else easelly